packages/aws-cdk-lib/aws-stepfunctions/lib/states/state.ts

```diff
@@ -487,6 +560,21 @@ export abstract class State extends Construct implements IChainable {
     };
   }
 
+  /**
+   * Render QueryLanguage in ASL JSON format if needed.
+   */
+  protected renderQueryLanguage(topLevelQueryLanguage?: QueryLanguage): any {
+    topLevelQueryLanguage = topLevelQueryLanguage ?? QueryLanguage.JSONPATH;
+    if (topLevelQueryLanguage === QueryLanguage.JSONATA && this.queryLanguage === QueryLanguage.JSONPATH) {
+      throw new Error(`'queryLanguage' can not be 'JSONPath' if set to 'JSONata' for whole state machine ${this.node.path}`);
+    }
+    const queryLanguage = topLevelQueryLanguage === QueryLanguage.JSONPATH && this.queryLanguage === QueryLanguage.JSONATA
+      ? QueryLanguage.JSONATA : undefined;
```

Comment:
Just want to double check that we can have the top level language to be JSON path but individual state to be JSONata?

Reply:
nit: I think we can simplify the highlighted part to prefer the state-level query language instead of the if check.

Reply:
OK! I thought small code is better than specify code in file size point of view. However, maybe simple code is better than currently!

Reply:
Ah, I re-thought this. This change makes breaking change for all step functions users, because all steps in definition are updated by this feature. We must avoid the breaking change.